sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/api/IdentityAbstractRestApiAction.java

```diff
@@ -0,0 +1,565 @@
+                case PUT:
+                    handlePut(channel, request, client, validator.getContentAsNode());
+                    break;
+                case GET:
+                    handleGet(channel, request, client, validator.getContentAsNode());
+                    break;
+                default:
+                    throw new IllegalArgumentException(request.method() + " not supported");
+            }
+        } catch (JsonMappingException jme) {
+            throw jme;
+            // TODO strip source
+            // if(jme.getLocation() == null || jme.getLocation().getSourceRef() == null) {
+            // throw jme;
+            // } else throw new JsonMappingException(null, jme.getMessage());
+        }
+    }
+
+    protected void handleDelete(final RestChannel channel, final RestRequest request, final Client client, final JsonNode content)
+        throws IOException {
+        final String name = request.param("username");
+
+        if (name == null || name.length() == 0) {
+            badRequestResponse(channel, "No " + getResourceName() + " specified.");
+            return;
+        }
+
+        final SecurityDynamicConfiguration<?> existingConfiguration = load(getConfigName());
+
+        if (!isWriteable(channel, existingConfiguration, name)) {
+            return;
+        }
+
+        boolean existed = existingConfiguration.exists(name);
+        existingConfiguration.remove(name);
+
+        if (existed) {
+            saveAnUpdateConfigs(
+                client,
+                request,
+                getConfigName(),
+                existingConfiguration,
+                new OnSucessActionListener<IndexResponse>(channel) {
+
+                    @Override
+                    public void onResponse(IndexResponse response) {
+                        successResponse(channel, "'" + name + "' deleted.");
+                    }
+                }
+            );
+
+        } else {
+            notFound(channel, getResourceName() + " " + name + " not found.");
+        }
+    }
+
+    protected void handlePut(final RestChannel channel, final RestRequest request, final Client client, final JsonNode content)
+        throws IOException {
+
+        final String name = request.param("name");
+
+        if (name == null || name.length() == 0) {
+            badRequestResponse(channel, "No " + getResourceName() + " specified.");
+            return;
+        }
+
+        final SecurityDynamicConfiguration<?> existingConfiguration = load(getConfigName());
+
+        if (existingConfiguration.getSeqNo() < 0) {
+            forbidden(
+                channel,
+                "Security index need to be updated to support '" + getConfigName().toLCString() + "'. Use SecurityAdmin to populate."
+            );
+            return;
+        }
+
+        if (!isWriteable(channel, existingConfiguration, name)) {
+            return;
+        }
+
+        if (isReadonlyFieldUpdated(existingConfiguration, content)) {
```

Comment:
I see that this function currently returns only `false`. How will this eventually be used?